src/lava/magma/compiler/compiler.py

```diff
@@ -206,6 +234,8 @@ def _expand_sub_proc_model(self,
 
         # Build SubProcessModel
         model = model_cls(proc)
+        # Check if VarPorts are configured and propagate them to the sub process
+        self._propagate_var_ports(proc)
```

Comment:
I would probably first check if there are VarPorts and only then call the function to propagate